design/metallb-operator.md

```diff
@@ -0,0 +1,62 @@
+# MetalLB Operator
+
+## Summary
+
+MetalLB currently includes [instructions and resources for deploying
+MetalLB](https://metallb.universe.tf/installation/) using a few methods:
+directly with manifests, customized manifests with kustomize, or via helm.
+Another approach for managing the deployment of software on Kubernetes is via
+an [operator](https://operatorhub.io/what-is-an-operator).
+
+This document proposes creating an operator for MetalLB in a new git repository
+under the metallb github organization and publishing it on
+[OperatorHub.io](https://operatorhub.io/).
+
+## Motivation
+
+For any Kubernetes environment that primarily uses operators to manage add-ons
+to the cluster, it would be beneficial to have the option of using an operator
+to manage MetalLB, as well.
+
+In [another proposal](https://github.com/metallb/metallb/pull/833), we have
+discussed creating a CRD based configuration interface for MetalLB. It seems
+that settling on the ideal data model may require some more time and
+experimentation. We can make use of this separate operator repo as a place to
+try out a CRD interface for MetalLB which writes out a ConfigMap and doesn't
+have to disrupt to core MetalLB project.
+
+### Goals
+
+- Create an operator in a new git repository that makes it easy to deploy
+  MetalLB on clusters where operators are the preferred method for managing
+  cluster add-ons.
+- Publish the operator on operatorhub.io to make it discoverable
+- Use the separate operator repository as a place to experiment with a CRD
```

Comment:
Let's say we find a good CRD data model in the new repo - what would we do then? "Port" the CRD code to the MetalLB repo and get rid of the operator?

Reply:
Yes, I imagine we could move the CRD handling code to MetalLB.  I imagine the operator would remain, though.  It would still be responsible for managing the controller deployment and the speaker daemonset.  It would do the equivalent of what helm does.